### PR TITLE
AccountIterator

### PR DESCRIPTION
**Pull request by bgptr**
GoMobile does not export `Accounts.Acc` property. This PR resolves this with an iterator. 
This feature is used in PR https://github.com/raedahgroup/dcrios/pull/533

### CODE REVIEW

File: accounts.go [hunk 1]
```diff
@@ -49,6 +49,15 @@ func (lw *LibWallet) GetAccountsRaw(requiredConfirmations int32) (*Accounts, err
 	}, nil
 }
 
+func (accounts *Accounts) Next() *Account {
+	if accounts.Iter.CurrentIndex < len(accounts.Acc) {
```
Comment (C-ollins) [inline accounts.go]: Can find where `accounts.Iter` was initialized. Do this code run without errors?

Comment (bgptr) [reply accounts.go]: Yes, it runs. 
I use it like this:

`func walletAccounts(confirmations: Int32) -> [DcrlibwalletAccount] {
        var accounts = [DcrlibwalletAccount]()
        do {
            let accountsRaw = try self.getAccountsRaw(confirmations)
            accountsRaw.iter = DcrlibwalletAccountIterator()
            
            while let account = accountsRaw.next() {
                accounts.append(account)
            }
            
            return accounts
        } catch let error {
            print("Error fetching wallet accounts: \(error.localizedDescription)")
            return accounts
        }
}`

Comment (bgptr) [reply accounts.go]: Commited a nicer implementation: cb25842

File: accounts.go [hunk 2]
```diff
@@ -49,6 +49,22 @@ func (lw *LibWallet) GetAccountsRaw(requiredConfirmations int32) (*Accounts, err
 	}, nil
 }
 
+func (accounts *Accounts) GetAccountIterator() *AccountIterator {
```
Comment (itswisdomagain) [inline accounts.go]: The following reads better as `accounts.Iterator` rather than `accounts.GetAccountIterator`:
```suggestion
func (accounts *Accounts) Iterator() *AccountIterator {
```

File: accounts.go [hunk 3]
```diff
@@ -49,6 +49,22 @@ func (lw *LibWallet) GetAccountsRaw(requiredConfirmations int32) (*Accounts, err
...
+	return &AccountIterator{
+		CurrentIndex: 0,
+		Acc:          accounts.Acc,
+	}
+}
+
+func (iter *AccountIterator) Next() *Account {
```
Comment (itswisdomagain) [inline accounts.go]:
```suggestion
func (accountsInterator *AccountIterator) Next() *Account {
```

File: types.go [hunk 4]
```diff
@@ -40,6 +40,11 @@ type Account struct {
 	ImportedKeyCount int32
 }
 
+type AccountIterator struct {
```
Comment (itswisdomagain) [inline types.go]:
```suggestion
type AccountsIterator struct {
```
Also, the properties in this struct do not need to be public.

File: types.go [hunk 5]
```diff
@@ -40,6 +40,11 @@ type Account struct {
...
+	CurrentIndex int
+	Acc          []*Account
```
Comment (itswisdomagain) [inline types.go]:
```suggestion
	accounts          []*Account
```

File: go.mod [hunk 6]
```diff
@@ -40,3 +40,5 @@ require (
 	golang.org/x/sys v0.0.0-20190422165155-953cdadca894 // indirect
 	google.golang.org/appengine v1.5.0 // indirect
 )
+
+go 1.13
```
Comment (itswisdomagain) [inline go.mod]: Please undo the changes to this file.

Comment (bgptr) [reply go.mod]: done. sorry!
